packages/core/src/system.ts

```diff
@@ -0,0 +1,50 @@
+import { ActorRef, ActorSystem, ActorSystemInfo } from './types.js';
+
+const children = new Map<string, ActorRef<any>>();
+
+let sessionIdIndex = 0;
+
+export interface Registry {
+  bookId(): string;
+  register(id: string, actor: ActorRef<any>): string;
+  get(id: string): ActorRef<any> | undefined;
+  free(id: string): void;
+}
+
+export const registry: Registry = {
+  bookId() {
+    return `x:${sessionIdIndex++}`;
+  },
+  register(id, actor) {
+    children.set(id, actor);
+    return id;
+  },
+  get(id) {
+    return children.get(id);
+  },
+  free(id) {
+    children.delete(id);
+  }
+};
+
+export function createSystem<T extends ActorSystemInfo>(): ActorSystem<T> {
+  const children = new Map<string, ActorRef<any>>();
+  const keyedActors = new Map<keyof T['actors'], ActorRef<any> | undefined>();
+
+  return {
+    register: (actorRef) => {
+      const id = `x:${sessionIdIndex++}`;
+      children.set(id, actorRef);
+      return id;
+    },
+    unregister: (actorRef) => {
+      children.delete(actorRef.id);
+    },
+    get: (key) => {
+      return keyedActors.get(key) as T['actors'][any];
+    },
+    set: (key, actorRef) => {
+      keyedActors.set(key, actorRef);
+    }
```

Comment:
what should happen when an actor with a given key is already registered?

Reply:
It overwrites the previous actor

Reply:
It feels to me that this might be an easy mistake to make and one that might be quite hard to debug if somebody accidentally overwrites the existing registered actor. 

On top of that - there is now no way to **clear** a registered actor. This can lead to memory leaks

Reply:
> On top of that - there is now no way to **clear** a registered actor. This can lead to memory leaks

Registered actors are unregistered when they are stopped. If an actor is not stopped, it won't be unregistered, but the actor not being stopped is a memory leak anyway.